File: Editorial/791_Custom_Sort_String/Python3/1_Custom_Comparator.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        # Create a list from s to allow modification (strings are immutable in Python)
        result = list(s)

        # Sort using the index in order as the key
        result.sort(key=lambda c: order.index(c) if c in order else len(order))

        # Concatenate the list into a string
        return ''.join(result)

File: Editorial/791_Custom_Sort_String/CPP/1_Custom_Comparator.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        // Since strings are mutable in C++, we can sort s in-place
        sort(s.begin(), s.end(), [&order](char c1, char c2) {
            return order.find(c1) < order.find(c2);
        });
        return s;
    }
};
